tests/integration/test_endpoints.py

```diff
@@ -219,6 +221,109 @@ def test_user_no_jwt(client):
     assert response.status_code == 401
 
 
+@pytest.mark.parametrize('input', [
+    # full definition
+    {
+        'type': 'gdpr',
+        'category': 'newsletter',
+        'status': 'accepted',
+        'timestamp': datetime.now(timezone('UTC')).isoformat(),
+        'valid_until': datetime.now(timezone('UTC')).isoformat(),
+        'message': 'I consent to the ToS',
+        'source': 'pytest'
+    },
+    # minimal
+    {
+        'type': 'cookie',
+        'category': 'preferences',
+        'status': 'rejected',
+    }
+])
+def test_create_consent(client, session, tokens, input):
+    """Create a new consent."""
+    response = client.post("/consent", json=input, headers={
+        'Authorization': f"Bearer {tokens['write']}",
+    })
+    assert response.status_code == 201
+    consent_id = response.json['id']
+    assert Consent.query.filter(Consent.id == consent_id).count() == 1
+
+
+@pytest.mark.parametrize('input', [
+    # incorrect type
+    {
+        'type': 'gpdr',
+        'category': 'newsletter',
+        'status': 'accepted',
+    },
+    # incorrect category for 'cookie' consent
+    {
+        'type': 'cookie',
+        'category': 'fumctiomal',
+        'status': 'accepted',
+    },
+    # incorrect status
+    {
+        'type': 'cookie',
+        'category': 'strictly_necessary',
+        'status': 'akcepted',
+    },
+])
+def test_create_consent_fail(client, session, tokens, input):
+    """Create a new consent."""
+    response = client.post("/consent", json=input, headers={
+        'Authorization': f"Bearer {tokens['write']}",
+    })
+    assert response.status_code == 422
+
+
+def test_get_consent(app, client, session, models, tokens):
+    """Retrieve user consent data based on given token."""
+    response = client.get("/consent", headers={
+        'Authorization': f"Bearer {tokens['read']}",
+    })
+    assert response.status_code == 200
+
+    user_id = jwt.decode(
+        tokens['read'],
+        app.config['RSA_PUBLIC_KEY'],
+        app.config['ALGORITHM'],
+    )['usr']
+
+    # Get expected consents - Consents with most recent timestamp value
+    # for each group of consents that have identical combination of user_id,
+    # type, and category
+    # NOTE: Intentionally using different logic here than in resource.py for
+    #       robustness
```

Comment:
This is very good - the comment could maybe be even more clear by stating that the test groups the consents programmatically in order to test expected functioning of the SQL query.

Reply:
Addressed in https://github.com/DD-DeCaF/iam/pull/25/commits/cd617de09c0463238c002f15b706e48ffe1016e4